### PR TITLE
Fix CredentialData URI parser

### PR DESCRIPTION
**Pull request by AdamVe**
This pull request enhances the robustness and correctness of the `CredentialData` URI parsing logic and improves the test coverage for edge cases. The most important changes include stricter validation of URI query parameters, decoding URL-encoded values, and adding new test cases to ensure proper behavior.

### Improvements to URI Parsing Logic:

* Added validation to ensure the query string is not null or empty, throwing a `ParseUriException` if it is. (`oath/src/main/java/com/yubico/yubikit/oath/CredentialData.java`, [oath/src/main/java/com/yubico/yubikit/oath/CredentialData.javaL72-R94](diffhunk://#diff-04ffceee9ab2818173da5d6014e983c68169e803b177ab04b4577304b3234764L72-R94))
* Introduced decoding for URL-encoded query parameter values using `URLDecoder`, with error handling for unsupported encoding or invalid URL encoding. (`oath/src/main/java/com/yubico/yubikit/oath/CredentialData.java`, [oath/src/main/java/com/yubico/yubikit/oath/CredentialData.javaL72-R94](diffhunk://#diff-04ffceee9ab2818173da5d6014e983c68169e803b177ab04b4577304b3234764L72-R94))
* Added validation to ensure the `secret` parameter is present and non-empty, throwing a `ParseUriException` if it is missing or invalid. (`oath/src/main/java/com/yubico/yubikit/oath/CredentialData.java`, [oath/src/main/java/com/yubico/yubikit/oath/CredentialData.javaL98-R122](diffhunk://#diff-04ffceee9ab2818173da5d6014e983c68169e803b177ab04b4577304b3234764L98-R122))

### Test Coverage Enhancements:

* Updated the `testParseUriGood` test to include cases with URL-encoded `issuer` and `accountName`, ensuring proper decoding and validation. (`oath/src/test/java/com/yubico/yubikit/oath/CredentialDataTest.java`, [oath/src/test/java/com/yubico/yubikit/oath/CredentialDataTest.javaL48-R68](diffhunk://#diff-43360648f63d6dafe978bf66fef45124293150e94bdbaf75fdbeaf1461a1b712L48-R68))
* Added a new test case, `testParseInvalidIssuerQuery`, to verify that an invalid `issuer` query parameter without a value throws a `ParseUriException`. (`oath/src/test/java/com/yubico/yubikit/oath/CredentialDataTest.java`, [oath/src/test/java/com/yubico/yubikit/oath/CredentialDataTest.javaR96-R100](diffhunk://#diff-43360648f63d6dafe978bf66fef45124293150e94bdbaf75fdbeaf1461a1b712R96-R100))

### CODE REVIEW

File: oath/src/main/java/com/yubico/yubikit/oath/CredentialData.java [hunk 1]
```diff
@@ -69,9 +71,27 @@ public static CredentialData parseUri(URI uri) throws ParseUriException {
...
+        throw new ParseUriException("Invalid query parameter: " + line);
+      }
+      String decodedValue = parts[1];
+      try {
+        decodedValue = URLDecoder.decode(decodedValue, "UTF-8");
+      } catch (IllegalArgumentException e) {
+        throw new ParseUriException("Invalid URL encoding in parameter value: " + decodedValue);
+      } catch (UnsupportedEncodingException e) {
+        throw new ParseUriException(
+            "UTF-8 encoding not supported while parsing value: " + decodedValue);
```
[the remark this answers is not in the document]
Comment (AdamVe) [reply oath/src/main/java/com/yubico/yubikit/oath/CredentialData.java:91]: Cannot use the mentioned overload because this code is also used in Android and the specific call requires API level 33 (The sdk has minSdk 19).
